js/src/layers/VectorTileLayer.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+const layer = require('./Layer.js');
+
+export class LeafletVectorTileLayerModel extends layer.LeafletLayerModel {
+  defaults() {
+    return {
+      ...super.defaults(),
+      _view_name: 'LeafletVectorTileLayerView',
+      _model_name: 'LeafletVectorTileLayerModel',
+      url: '',
+    };
+  }
+}
+
+export class LeafletVectorTileLayerView extends layer.LeafletLayerView {
+   create_obj() {
+        var options = {
+            ...this.get_options(),
+            rendererFactory: L.canvas.tile,
+            }
+        this.obj = L.vectorGrid.protobuf(this.model.get('url'), options);
+        this.model.on('msg:custom', this.handle_message.bind(this));
+    }
+
+   model_events() {
+    super.model_events();
+    this.listenTo(
+      this.model,
+      'change:url',
+      function() {
+        this.obj.setUrl(this.model.get('url'), true);
+        this.obj.refresh();
+      },
+      this
+    );
+  }
+
+  handle_message(content) {
+    if (content.msg == 'redraw') {
+      this.obj.redraw();
+    }
+  }
```

Comment:
You also need to listen to options changes, and apply them. In WMSLayer there is a `setParam` method, I guess you have something equivalent in the Leaflet.VectorGrid.Protobuf

Reply:
Actually, as I see there is code there no way to update options and do a soft refresh in Leaflet.VectorGrid.Protobuf:- https://github.com/Leaflet/Leaflet.VectorGrid/issues/159

Reply:
I see. I guess we can try using `setOptions`. But let's not block this PR any longer.